include/openssl/bio.h

```diff
@@ -156,10 +156,12 @@ extern "C" {
 
 # ifndef OPENSSL_NO_KTLS
 #  define BIO_get_ktls_send(b)         \
-     (BIO_method_type(b) == BIO_TYPE_SOCKET \
+     ((BIO_method_type_final(b) == BIO_TYPE_SOCKET \
+       || BIO_method_type_final(b) == BIO_TYPE_FD) \
```

Comment:
I don't understand the addition of BIO_TYPE_FD. If you have BIO_TYPE_FD then you are using BIO_s_fd() - which does not understand the BIO_CTRL_GET_KTLS_SEND and BIO_CTRL_GET_KTLS_RECV ctrls.

Reply:
obviously some test cases are missing?

Reply:
Sorry, you are right. I got carried away and forgot the meaning of BIO_TYPE_FD.
Overall it should be possible to add the KTLS calls to this BIO type, but the fix is not related to it.